eureka-client-archaius2/src/main/java/com/netflix/discovery/guice/EurekaClientModule.java

```diff
@@ -0,0 +1,62 @@
+package com.netflix.discovery.guice;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.binder.LinkedBindingBuilder;
+import com.google.inject.name.Names;
+import com.netflix.appinfo.providers.EurekaInstanceConfigFactory;
+
+/**
+ * How to use:
+ *  - install this module to access all eureka client functionality.
+ *
+ * Custom config namespace may be registered as follows:
+ * <code>
+ * InjectorBuilder.fromModules(new EurekaClientModule() {
+ *      protected void configureEureka() {
+ *          bindEurekaInstanceConfigNamespace().toInstance("namespaceForMyInstanceConfig");
+ *          bindEurekaClientConfigNamespace().toInstance("namespaceForMyClientAndTransportConfig");
+ *      }
+ * }).createInjector()
+ * </code>
+ *
+ * This module support the binding of a custom {@link EurekaInstanceConfigFactory} to supply your own
+ * way of providing a config for the creation of an {@link com.netflix.appinfo.InstanceInfo} used for
+ * eureka registration.
+ *
+ * Custom {@link EurekaInstanceConfigFactory} may be registered as follows:
+ * <code>
+ * InjectorBuilder.fromModules(new EurekaClientModule() {
+ *      protected void configureEureka() {
+ *          bindEurekaInstanceConfigFactory().to(MyEurekaInstanceConfigFactory.class);
+ *      }
+ * }).createInjector()
+ * </code>
+ *
+ * Note that this module is NOT compatible with the archaius1 based {@link com.netflix.discovery.guice.EurekaModule}
```

Comment:
Pretty difficult to distinguish this module from the other. Slightly different names, same packages. Bound to confuse someone.